src/Downloader/Downloader.php

```diff
@@ -96,7 +96,19 @@ public function flush(?callable $callback = null): void
 
         $this->requests = [];
 
-        $this->client->pool($requests, function (Response $response) use ($callback): void {
+        foreach ($requests as $key => $request) {
+            if ($request->getResponse() !== null) {
+                $this->onResponseReceived($request->getResponse(), $callback);
+
+                unset($requests[$key]);
+            }
+        }
+
+        if (!\count($requests)) {
```

Comment:
I prefer to be explicit instead of relying on type coercion. So either `\count($requests) === 0` or `\empty($requests)`.